lib/jsxgettext.js

```diff
@@ -84,12 +84,28 @@ function getTranslatable(node, options) {
   if (options.keyword.indexOf(funcName) === -1)
     return false;
 
+  // Domain is not used during extraction, so ignore it
+  if(funcName.substr(0, 1) === "d"){
+    funcName = funcName.substr(1, funcName.length);
+    node.arguments.splice(0, 1);
+    arg = node.arguments[0];
+  }
+
+  // Always add the funcName as last element in return array
+  // If the gettext function's name starts with "np" (i.e. npgettext or np_) and its 3 arguments are strings, we regard it as context  
+  if (arg && funcName.substr(0, 2) === "np" && (isStrConcatExpr(arg) || isStringLiteral(arg)) && node.arguments[1] && (isStrConcatExpr(node.arguments[1]) || isStringLiteral(node.arguments[1])) && node.arguments[2] && (isStrConcatExpr(node.arguments[2]) || isStringLiteral(node.arguments[2])))
+    return [arg, node.arguments[1], node.arguments[2], funcName];
+
   // If the gettext function's name starts with "n" (i.e. ngettext or n_) and its first 2 arguments are strings, we regard it as a plural function
   if (arg && funcName.substr(0, 1) === "n" && (isStrConcatExpr(arg) || isStringLiteral(arg)) && node.arguments[1] && (isStrConcatExpr(node.arguments[1]) || isStringLiteral(node.arguments[1])))
-    return [arg, node.arguments[1]];
-
+    return [arg, node.arguments[1], funcName];
+  
+  // If the gettext function's name starts with "p" (i.e. pgettext or p_) and its 2 arguments are strings, we regard it as context  
```

Comment:
There's now a lot of repetition around argument type checks here so I think we should move them into their own function. Something like:
```
function areArgsString(args) {
    // magic
    return wand;
}
```